addons/mrp/tests/test_order.py

```diff
@@ -5239,6 +5239,25 @@ def test_product_qty_digits_precision(self):
         self.assertEqual(unbuild_order.state, 'done')
         self.assertEqual(unbuild_order.product_qty, 1.23456)
 
+    def test_workorders_without_worcenter_planned_slots(self):
+        mo = self.env['mrp.production'].create({
+            'product_id': self.product_6.id,
+            'product_qty': 1.0,
+            'bom_id': self.bom_3.id,
+        })
+        mo.action_confirm()
+
+        first_wo, second_wo, third_wo = mo.workorder_ids
+
+        first_wo.button_start()
+        mo.workorder_ids = [Command.update(second_wo.id, {"duration": 12.0})]
```

Comment:
As I already suggested.
```suggestion
       first_wo.button_start()
        with Form(mo) as mo_form:
            with mo_form.workorder_ids.edit(1) as second_wo_line:
                second_wo_line.duration = 12.0
```